ow_plexil/src/plexil-adapter/OwlatInterface.cpp

```diff
@@ -5,34 +5,104 @@
+const string Name_OwlatArmMoveCartesian =   "/owlat_sim/ARM_MOVE_CARTESIAN";
+const string Name_OwlatArmMoveCartesianGuarded =   "/owlat_sim/ARM_MOVE_CARTESIAN_GUARDED";
+const string Name_OwlatArmMoveJoint =   "/owlat_sim/ARM_MOVE_JOINT";
+const string Name_OwlatArmMoveJoints =   "/owlat_sim/ARM_MOVE_JOINTS";
+const string Name_OwlatArmMoveJointsGuarded =   "/owlat_sim/ARM_MOVE_JOINTS_GUARDED";
+const string Name_OwlatArmPlaceTool =   "/owlat_sim/ARM_PLACE_TOOL";
+const string Name_OwlatArmSetTool =   "/owlat_sim/ARM_SET_TOOL";
+const string Name_OwlatArmStop =   "/owlat_sim/ARM_STOP";
+const string Name_OwlatArmTareFS =   "/owlat_sim/ARM_TARE_FS";
+const string Name_OwlatTaskDropoff =   "/owlat_sim/TASK_DROPOFF";
+const string Name_OwlatTaskPSP =   "/owlat_sim/TASK_PSP";
+const string Name_OwlatTaskScoop =   "/owlat_sim/TASK_SCOOP";
+const string Name_OwlatTaskShearBevameter =   "/owlat_sim/TASK_SHEAR_BEVAMETER";
 
 // Used as indices into the subsequent vector.
 enum LanderOps {
   OwlatUnstow,
-  OwlatStow
+  OwlatStow,
+  OwlatArmMoveCartesian,
+  OwlatArmMoveCartesianGuarded,
+  OwlatArmMoveJoint,
+  OwlatArmMoveJoints,
+  OwlatArmMoveJointsGuarded,
+  OwlatArmPlaceTool,
+  OwlatArmSetTool,
+  OwlatArmStop,
+  OwlatArmTareFS,
+  OwlatArmTaskDropoff,
+  OwlatArmTaskPSP,
+  OwlatArmTaskScoop,
+  OwlatArmTaskShearBevameter
 };
 
-static std::vector<string> LanderOpNames = {
-  Name_OwlatUnstow,
-  Name_OwlatStow
-};
+static std::vector<string> LanderOpNames =
+  { Name_OwlatUnstow,
+    Name_OwlatStow,
+    Name_OwlatArmMoveCartesian,
+    Name_OwlatArmMoveCartesianGuarded,
+    Name_OwlatArmMoveJoint,
+    Name_OwlatArmMoveJoints,
+    Name_OwlatArmMoveJointsGuarded,
+    Name_OwlatArmPlaceTool,
+    Name_OwlatArmSetTool,
+    Name_OwlatArmStop,
+    Name_OwlatArmTareFS,
+    Name_OwlatTaskDropoff,
+    Name_OwlatTaskPSP,
+    Name_OwlatTaskScoop,
+    Name_OwlatTaskShearBevameter
+  };
 
+// Task Shear Bevameter Callback
+static double BevameterStopReasonVar = 0;
+template<int OpIndex, typename T>
+static void task_shear_bevameter_done_cb
+(const actionlib::SimpleClientGoalState& state,
+ const T& result)
+{
+  ROS_INFO("Shear Bevameter Stop Reason: : %d", result->stop_reason.value);
+  BevameterStopReasonVar = result->stop_reason.value;
+  publish("ShearBevameterStopReason", BevameterStopReasonVar);
+  ROS_INFO ("/owlat_sim/TASK_SHEAR_BEVAMETER finished in state %s", 
+            state.toString().c_str());
+}
 
-std::shared_ptr<OwlatInterface> OwlatInterface::m_instance = nullptr;
+// Task PSP Callback
+static double PSPStopReasonVar = 0;
+template<int OpIndex, typename T>
+static void task_psp_done_cb
+(const actionlib::SimpleClientGoalState& state,
+ const T& result)
+{
+  ROS_INFO("PSP Stop Reason: : %d", result->stop_reason.value);
+  PSPStopReasonVar = result->stop_reason.value;
+  publish("PSPStopReason", PSPStopReasonVar);
+  ROS_INFO ("/owlat_sim/TASK_PSP finished in state %s", 
+            state.toString().c_str());
+}
```

Comment:
Why aren't these member variables and member functions like the other callbacks?

Reply:
The action callbacks usually use the template class in action_support but these needed a custom callback for the response from the action servers. I went off of OwInterface.cpp where Mike also used static functions vs member functions when overriding the template class. 

@kmdalal probably has better insight on this.

Reply:
This looks okay to me.  Static functions were originally used for the callbacks for their simplicity.  Overriding the defaults with custom static functions follows the existing pattern.